lib/Jsrt/Core/JsrtCore.cpp

```diff
@@ -1460,7 +1461,8 @@ CHAKRA_API JsIsCallable(_In_ JsValueRef object, _Out_ bool *isCallable)
         VALIDATE_INCOMING_OBJECT(object, scriptContext);
         PARAM_NOT_NULL(isCallable);
 
-        *isCallable = Js::JavascriptConversion::IsCallable(object);
+        *isCallable = (Js::JavascriptConversion::IsCallable(object) && !Js::JavascriptOperators::IsClassConstructor(object)) ||
+                       Js::VarIs<Js::JavascriptGeneratorFunction>(object);
```

Comment:
`JavascriptOperators::IsClassConstructor` really should not be returning positive for generator functions - editing it so it doesn't; seems better than an extra check here - I tried offline and it doesn't appear to break anything...

Reply:
Maybe it's problem that is reproducable at Windows machine, since i ran some tests that show that `IsClassConstructor` shows a false-positive:  
Format: (checks)  
(result with removed VarIs check) (checked value)
```
IsCallable: true, IsClassConstructor: false, VarIs<JavascriptGeneratorFunction>: false
true () => { }
IsCallable: true, IsClassConstructor: true, VarIs<JavascriptGeneratorFunction>: true
false async () => { }
IsCallable: true, IsClassConstructor: false, VarIs<JavascriptGeneratorFunction>: false
true function () { }
IsCallable: true, IsClassConstructor: true, VarIs<JavascriptGeneratorFunction>: true
false function* () { }
IsCallable: true, IsClassConstructor: true, VarIs<JavascriptGeneratorFunction>: true
false async function () { }
IsCallable: true, IsClassConstructor: true, VarIs<JavascriptGeneratorFunction>: true
false async function* () { }
```
But i agree that `VarIs` check is not required to fix the initial problem.

Reply:
So, should i move `VarIs` check to `IsClassConstructor`?

Reply:
I thought the false positive was because of the second case over in JavascriptOperators::IsClassConstructor

As a generator function is neither a ScriptFunction nor an external function. BUT I can't see why we're checking for those there.

If we delete that second case should then be able to get rid of the extra check here.